File: python-version/LC3354/count_valid_selections.py
from typing import List


class Solution:
    @staticmethod
    def countValidSelections(nums: List[int]) -> int:
        sm = sum(nums)
        cur, ans = 0, 0
        for num in nums:
            cur += num
            if num == 0 and cur * 2 == sm:
                ans += 2
            if num == 0 and abs(cur * 2 - sm) == 1:
                ans += 1
        return ans

File: go-version/leetcode/LC3354/count_valid_selections.go
package LC3354

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/10/28 下午11:28
 * @FilePath: count_valid_selections
 * @Description: 前缀和进行分解判断即可
 */

func countValidSelections(nums []int) int {
	var sum, cur, ans int
	for _, num := range nums {
		sum += num
	}
	for _, num := range nums {
		cur += num
		if num == 0 && cur*2 == sum {
			ans += 2
		}
		if num == 0 && abs(cur*2-sum) == 1 {
			ans += 1
		}
	}
	return ans
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
